src/app/pages/instruction-viewer/instruction-viewer.component.ts

```diff
@@ -528,52 +529,58 @@ export class InstructionViewerComponent implements OnInit, OnDestroy {
         const camera: OrthographicCamera = scene.userData["camera"];
         // get its position relative to the page's viewport
         const rect = element.getBoundingClientRect();
+        const rectParent = element.parentElement!.getBoundingClientRect();
         const canvasrect = this.canvas.getBoundingClientRect();
-        rect.x -= canvasrect.x;
-        // set the viewport
-        const width = rect.right - rect.left;
-        const height = rect.bottom - rect.top;
-        const left = rect.left;
-        const bottom = this.canvas.clientHeight - rect.bottom;
-        this.renderer.setViewport(left, bottom, width, height);
-        this.renderer.setScissor(left, bottom, width, height);
+        const top = rect.y - canvasrect.y;
+        const left = rect.x - canvasrect.x;
+        console.log(left, rect.height - top, rect.width, rect.height);
+        this.renderer.setSize(rectParent.width, rectParent.height);
+
+        this.renderer.setViewport(0, 0, canvasrect.width, canvasrect.height);
+        this.renderer.setScissor(0, 0, canvasrect.width, canvasrect.height);
+        this.renderer.clearColor();
+        this.renderer.clearDepth();
+
+        this.renderer.setViewport(left, top, rect.width, rect.height);
+        this.renderer.setScissor(left, top, rect.width, rect.height);
         this.renderer.render(scene, camera);
 
         //render partList n stuff
         for (let i = 0; i < this.partListScenes.length; i++) {
           const scene = this.partListScenes[i];
           const element: HTMLDivElement = scene.userData["element"];
-          const camera: OrthographicCamera = scene.userData["camera"];
+          // const camera: OrthographicCamera = scene.userData["camera"];
 
           // get its position relative to the page's viewport
           const rect = element.getBoundingClientRect();
+          const top = rect.top - canvasrect.top;
+          const left = rect.left - canvasrect.left;
           // skip if offscreen
-          if (rect.bottom < 0 || rect.top > this.renderer?.domElement.clientHeight ||
-            rect.right < 0 || rect.left > this.renderer.domElement.clientWidth)
-            continue;
+          // if (rect.bottom < 0 || rect.top > this.renderer?.domElement.clientHeight ||
+          //     rect.right < 0 || rect.left > this.renderer.domElement.clientWidth)
+          //     continue;
 
           //this.updateControls(this.mainCamera,this.target,this.cameraCoordinates,this.scrollDelta,this.panDelta,this.dragDelta); TODO add for the parts
 
           // set the viewport
-          const width = rect.right - rect.left;
-          const height = rect.bottom - rect.top;
-          const left = rect.left;
-          const bottom = this.renderer.domElement.clientHeight - rect.bottom;
-          this.renderer.setViewport(left, bottom, width, height);
-          this.renderer.setScissor(left, bottom, width, height);
+          this.renderer.setViewport(rect.left, 0, rect.width, 96);
```

Comment:
here i hardcoded the height to 96 which matches what should be there, i tried to fix the divs, but i was unable to do it quickly so i leave it to you, should be easy, you can also hardcode it